selfdrive/debug/internal/measure_steering_accuracy.py

```diff
@@ -16,66 +17,112 @@ def sigint_handler(signal, frame):
+        angle_error = abs(desired_angle - actual_angle)
         actual_angle = round(actual_angle, 1)
         desired_angle = round(desired_angle, 1)
         angle_error = round(angle_error, 2)
         angle_abs = int(abs(round(desired_angle, 0)))
 
-        # collect stats
-        stats[angle_abs]["err"] += angle_error
-        stats[angle_abs]["cnt"] += 1
-        if actual_angle == desired_angle:
-          stats[angle_abs]["="] += 1
-        else:
-          if desired_angle == 0.:
-            overshoot = True
-          else:
-            overshoot = desired_angle < actual_angle if desired_angle > 0. else desired_angle > actual_angle
-          stats[angle_abs]["+" if overshoot else "-"] += 1
+        for group, group_props in all_groups.items():
+          if v_ego > group_props[0]:
+            # collect stats
+            speed_group_stats[group][angle_abs]["cnt"] += 1
+            speed_group_stats[group][angle_abs]["err"] += angle_error
+            speed_group_stats[group][angle_abs]["steer"] += abs(steer)
+            if len(d_path_points):
+              speed_group_stats[group][angle_abs]["dpp"] += abs(d_path_points[0])
+            if control_state.saturated:
+              speed_group_stats[group][angle_abs]["saturated"] += 1
+            if actual_angle == desired_angle:
+              speed_group_stats[group][angle_abs]["="] += 1
+            else:
+              if desired_angle == 0.:
+                overshoot = True
+              else:
+                overshoot = desired_angle < actual_angle if desired_angle > 0. else desired_angle > actual_angle
+              speed_group_stats[group][angle_abs]["+" if overshoot else "-"] += 1
+            break
     else:
       cnt = 0
 
     if msg_cnt % 100 == 0:
       print(chr(27) + "[2J")
       if cnt != 0:
-        print("COLLECTING ...")
+        print("COLLECTING ...\n")
       else:
-        print("DISABLED (speed too low, not engaged, or steer override)")
-      for k in sorted(stats.keys()):
-        v = stats[k]
-        print(f'angle: {k:#2} | error: {round(v["err"] / v["cnt"], 2):2.2f} | =:{int(v["="] / v["cnt"] * 100):#3}% | +:{int(v["+"] / v["cnt"] * 100):#4}% | -:{int(v["-"] / v["cnt"] * 100):#3}% | count: {v["cnt"]:#4}')
+        print("DISABLED (not active, standstill, steering override, or lane change)\n")
+      for group in display_groups:
+        if len(speed_group_stats[group]) > 0:
+          print(f"speed group: {group:10s} {all_groups[group][1]:>91s}")
+          print(f"  {'-'*113}")
+          for k in sorted(speed_group_stats[group].keys()):
+            v = speed_group_stats[group][k]
+            print(f'  {k:#2}° | actuator:{int(v["steer"] / v["cnt"] * 100):#3}% | error: {round(v["err"] / v["cnt"], 2):2.2f}° | =:{int(v["="] / v["cnt"] * 100):#3}% | +:{int(v["+"] / v["cnt"] * 100):#4}% | -:{int(v["-"] / v["cnt"] * 100):#3}% | sat cnt: {v["saturated"]:#4} | dpathpoints: {round(v["dpp"] / v["cnt"], 2):2.2f} | total: {v["cnt"]:#5}')
```

Comment:
Dealers choice but personally: " -, =, + " column order was a little more readable. 

```suggestion
            print(f'  {k:#2}° | actuator:{int(v["steer"] / v["cnt"] * 100):#3}% | error: {round(v["err"] / v["cnt"], 2):2.2f}° | -:{int(v["-"] / v["cnt"] * 100):#3}% | =:{int(v["="] / v["cnt"] * 100):#3}% | +:{int(v["+"] / v["cnt"] * 100):#3}% | sat cnt: {v["saturated"]:#4} | dpathpoints: {round(v["dpp"] / v["cnt"], 2):2.2f} | total: {v["cnt"]:#5}')
```

Reply:
agreed